### DISCUSSION TITLE
Add the ability to move the menu to another screen location or another monitor

### DISCUSSION THREAD

**Discussion by LintStalker** [YimMenu/YimMenu#1686] [Ideas]
Could you add the ability to move the menu to another location on the screen, or to another monitor?  It's a little annoying that when you are tweaking a vehicle, it's hidden by the menu.

**Comment by Yimura** [comment]
How small is the resolution that the game is running on for you?

**Reply by Yimura** [reply]
This is how my game looks for example, I know it's an ultra-wide but a generic 1080p screen should also be fine.

![image](https://github.com/YimMenu/YimMenu/assets/24669514/dd404a1b-abe0-45a6-b606-294001173382)

**Reply by LintStalker** [reply]
My screen resolution is 1920 X 1080   I haven't changed the menu size

**Reply by LintStalker** [reply]
Hmmmmm....  I will have to take a look at that, and get back

**Reply by Yimura** [reply]
The problem is that it’s basically impossible for us to render outside of GTA, and moving YimMenu outside of the game would kinda make it useless since you need to tab out to use it.

**Reply by Yimura** [reply]
Well it’s possible for us to render outside of the game by creating a second window and rendering there but that would require a bunch of changes to the rendering pipeline to support both hijacking the game’s rendering and allowing for external rendering.

**Reply by LintStalker** [reply]
fair enough, just thought I would ask.  It's an awesome menu by the way!  You have done a great job!   

Any chance you could add the ability to set a vehicle as a Pegasus vehicle so that I could spawn planes and helicopters and bring them into my hanger?

**Reply by tupoy-ya** [reply]
> The problem is that it’s basically impossible for us to render outside of GTA, and moving YimMenu outside of the game would kinda make it useless since you need to tab out to use it.

It is possible, it's even implemented in ImGui docking branch, and in TupoyeMenu.
The problem is, it's not working very well - for example: Dragging an ImGui window outside the game will make your mouse stuck to that window and you will have to do some weird alt tabbing to fix it.
Maybe it's just a wine problem but i have it disabled because it's not very stable.
And it also closes all the extra windows if you close the in game UI.

https://github.com/TupoyeMenu/TupoyeMenu/blob/7284e9401d491ebf660496853f2cc8c20af68165/src/renderer.cpp#L55

**Reply by xiaoxiao921** [reply]
Internal injected DLL can still render to a new window and have it overlay like. This is c# but porting this to c++ should be trivial https://github.com/zaafar/ClickableTransparentOverlay/tree/master

**Reply by Yimura** [reply]
Yeah but there’s a problem you see, how would we prevent the game from holding the mouse cursor within its bounds?

**Reply by LintStalker** [reply]
Ok, no problem.  Probably not worth it if it's a lot of work.  Better to work on other, more important features.  I need to look into why my resolution is so low on my system.

**Reply by xiaoxiao921** [reply]
That's only doable through win api calls so shouldn't be hard to modify this behavior @Yimura

**Reply by maybegreat48** [reply]
> Any chance you could add the ability to set a vehicle as a Pegasus vehicle so that I could spawn planes and helicopters and bring them into my hanger?

That's considered recovery, so no

**Comment by LintStalker** [comment]
Why don’t you want to add recoveries to the menu?

**Comment by LintStalker** [comment]
That makes sense, you definitely don't want R* after you.

Do you care if recoveries are added via scripting?  Would R* still go after
you for what someone added on their own?

**Reply by Yimura** [reply]
People are free to do with the scripting API, we have given all the tools you need to modify the game’s memory. Obviously no warranty is given if you get banned.

**Comment by LintStalker** [comment]
That's awesome! Is there a place where people can share scripts?  Maybe a
thread on unknowncheats could be created, like like the script thread for
Kiddions modest menu?  It would be very helpful to folks like me that suck
at coding.
